tests/ops/qubit/test_matrix_ops.py

```diff
@@ -576,6 +578,114 @@ def test_matrix_already_in_cache_list(self, mat, op):
+        qml.BlockEncode(input_matrix, wires)
+
+    @pytest.mark.parametrize(
+        ("input_matrix", "wires"),
+        [(1, 1), (1, 2), (1, [1]), (1, range(2)), (np.identity(2), ["a", "b"])],
+    )
+    def test_varied_wires(self, input_matrix, wires):
+        qml.BlockEncode(input_matrix, wires)
+
+    @pytest.mark.parametrize(
+        ("input_matrix", "wires", "msg"),
+        [
+            (
+                [[0, 1], [1, 0]],
+                1,
+                f"Block encoding a ({2} x {2}) matrix requires a hilbert space of size"
+                f" at least ({4} x {4}). Cannot be embedded in a {1} qubit system.",
+            ),
+        ],
+    )
+    def test_correct_error_message(self, input_matrix, wires, msg):
+        with pytest.raises(ValueError, match=msg):
+            qml.BlockEncode(input_matrix, wires)
+
+    @pytest.mark.parametrize(
+        ("input_matrix", "wires", "output_matrix"),
+        [
+            (1, 0, [[1, 0], [0, -1]]),
+            (0.3, 0, [[0.3, 0.9539392], [0.9539392, -0.3]]),
+            (
+                [[0.1, 0.2], [0.3, 0.4]],
+                range(2),
+                [
+                    [0.1, 0.2, 0.97283788, -0.05988708],
+                    [0.3, 0.4, -0.05988708, 0.86395228],
+                    [0.94561648, -0.07621992, -0.1, -0.3],
+                    [-0.07621992, 0.89117368, -0.2, -0.4],
+                ],
+            ),
+            (
+                0.1,
+                range(2),
+                [[0.1, 0.99498744, 0, 0], [0.99498744, -0.1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]],
+            ),
+        ],
+    )
+    def test_correct_output_matrix(self, input_matrix, wires, output_matrix):
+        assert np.allclose(qml.matrix(qml.BlockEncode)(input_matrix, wires), output_matrix)
+
+    @pytest.mark.parametrize(
+        ("input_matrix", "wires"),
+        [
+            (1, 0),
+            (0.3, 0),
+            (np.array([[0.1, 0.2], [0.3, 0.4]]), range(2)),
+            (np.array([[0.1, 0.2, 0.3]]), range(2)),
+            (np.array([[0.1], [0.2], [0.3]]), range(2)),
+            (np.array([[0.1, 0.2], [0.3, 0.4], [0.5, 0.6]]), range(3)),
+            (np.array([[1, 2], [3, 4]]), range(2)),
+        ],
+    )
+    def test_unitary(self, input_matrix, wires):
+        mat = qml.matrix(qml.BlockEncode(input_matrix, wires))
+        assert np.allclose(np.eye(len(mat)), mat.dot(mat.T.conj()))
+
+    @pytest.mark.parametrize(
+        ("device_type", "wires", "input_matrix"),
+        [
+            ("default.qubit", range(1), pnp.array(0.3)),
+            ("default.qubit", range(2), pnp.diag([0.1, 0.2])),
+            ("default.qubit.autograd", range(1), pnp.array(0.5)),
+            ("default.qubit.autograd", range(2), pnp.diag([0.2, 0.3])),
+            ("default.qubit.tf", range(1), pnp.array(0.4)),
+            ("default.qubit.tf", range(2), pnp.diag([0.3, 0.5])),
+        ],
+    )
+    def test_differentiability(self, device_type, wires, input_matrix):
```

Comment:
We have this pytest marker for each interface. I would split up the gradient tests so that there is one for each interface. Also not 100% blocking, I think we can get away with just supporting autograd and jax to start with, but ideally we should have this for all interfaces. 

Also there are multiple gradient methods, it should be worth checking if it is differentiable with each one, atleast for the autograd interface. If it isn't compatible, then atleast we have a test for it that we can return to once we decide to support it.